contractcourt/channel_arbitrator.go

```diff
@@ -1131,11 +1139,102 @@ func (c *ChannelArbitrator) sweepAnchors(anchors []*lnwallet.AnchorResolution,
 		if err != nil {
 			return err
 		}
+
+		return nil
+	}
+
+	// Calculate the deadline. This is only valid for sweeping the local
+	// anchor.
+	deadline, err := c.findCommitmentDeadline(heightHint)
+	if err != nil {
+		return err
+	}
+
+	for _, anchor := range anchors {
+		// The first item is the anchor of the local commitment tx, we
+		// will use the deadline calcuated from the active HTLCs.
+		if err := sweepWithDeadline(anchor, deadline); err != nil {
+			return err
+		}
+
+		// For the rest anchors, use the default conf target.
```

Comment:
I'm not sure if this is what we want. If we decided it is urgent that we force close the channel, the same urgency should apply to all anchors.

Reply:
IIRC, the current logic (us attempting to anchor down all three) attempts to account for the case that by the time _we _ force close, one of the remote party's commitments is already in the mempool, so then we attempt to try to anchor down that as well. The issue ofc, is that since we don't watch the mempool, we can't tell if they're attempting to anchor it down themselves, by looking at the transaction spending _their_ anchor output.

Reply:
> we can only learn that they've been broadcasted after they are confirmed and to be found in a block

We always assume that _one of the three_ valid commitments are in the mempool, but that we don't know which one. Because of this we anchor down all three with the same urgency.